querybook/server/logic/elasticsearch.py

```diff
@@ -263,33 +298,37 @@ def get_query_cells_iter(batch_size=1000, session=None):
 
 
 @with_session
-def query_cell_to_es(query_cell, session=None):
-    query_cell_id = query_cell.id
+def query_cell_to_es(query_cell, fields=None, session=None):
     query_cell_meta = query_cell.meta
-
-    engine_id = query_cell_meta.get("engine")
-    engine = get_query_engine_by_id(engine_id, session=session)
-
     query = query_cell.context
-    table_names, _ = process_query(query, language=(engine and engine.language))
-    table_names = list(chain.from_iterable(table_names))
-
     datadoc = query_cell.doc
 
-    expand_query = {
-        "id": query_cell_id,
-        "query_type": "query_cell",
-        "title": query_cell_meta.get("title", "Untitled"),
-        "data_doc_id": datadoc and datadoc.id,
-        "environment_id": datadoc and datadoc.environment_id,
-        "author_uid": datadoc and datadoc.owner_uid,
-        "engine_id": engine_id,
-        "statement_type": get_table_statement_type(query),
-        "created_at": DATETIME_TO_UTC(query_cell.created_at),
-        "full_table_name": table_names,
-        "query_text": query,
+    engine_id = query_cell_meta.get("engine")
+    engine = admin_logic.get_query_engine_by_id(engine_id, session=session)
+
+    def get_table_names():
```

Comment:
can you make get_table_names into a helper function that is shared by execution and cell